### ISSUE TITLE
Upstream Merge

### ISSUE THREAD

**Issue by bbrockman** [sscpac/chat-locker#71]
- [x] Review differences from upstream merge
- [x] See if it is worth the pain of merging
- [x] Merge if required

**Comment by rwakida** [comment]
Reasons to merge:
1. Administrative UI settings/pages are updated. 
2. Able to disable edit/delete messages in a conversation, disable new user registration, disable email verification
3. Adds statistics, users, rooms information. 
4. Able to customize home page title/body via administration settings. 
5. Remove channel/private group name restrictions.  I added this in myself as a side effect of their routing changes.  It is NOT supported by RocketChat.  We slugify the name and use it as the id.  The raw name is saved as a separate displayName field.  Only restriction is that we can't name a channel/group if the slugified value already exists.  e.g. pg1.1 and pg1 1 would have the same slug value of pg1-1
6. They added system backend methods that I requested for creating channels etc.
